app/src/pages/settings/AnnotationConfigDialog.tsx

```diff
@@ -0,0 +1,424 @@
+import React from "react";
+import { Controller, useFieldArray, useForm } from "react-hook-form";
+import { css } from "@emotion/react";
+
+import { Card } from "@arizeai/components";
+
+import {
+  Button,
+  Dialog,
+  FieldError,
+  Flex,
+  Form,
+  Icon,
+  Icons,
+  Input,
+  Label,
+  NumberField,
+  Radio,
+  RadioGroup,
+  TextArea,
+  TextField,
+  View,
+} from "@phoenix/components";
+import { useNotifyError, useNotifySuccess } from "@phoenix/contexts";
+import {
+  AnnotationConfig,
+  AnnotationConfigCategorical,
+  AnnotationConfigContinuous,
+  AnnotationConfigFreeform,
+  AnnotationConfigOptimizationDirection,
+  AnnotationConfigType,
+} from "@phoenix/pages/settings/types";
+
+const optimizationDirections = [
+  "MAXIMIZE",
+  "MINIMIZE",
+] satisfies AnnotationConfigOptimizationDirection[];
+
+const types = [
+  "CATEGORICAL",
+  "CONTINUOUS",
+  "FREEFORM",
+] satisfies AnnotationConfigType[];
+
+export const AnnotationConfigDialog = ({
+  onAddAnnotationConfig,
+  initialAnnotationConfig,
+}: {
+  onAddAnnotationConfig: (
+    config: AnnotationConfig,
+    {
+      onCompleted,
+      onError,
+    }?: { onCompleted?: () => void; onError?: (error: string) => void }
+  ) => void;
```

Comment:
this callback is kinda confusing, why not just two callbacks? It would keep the interface a bit more intuitive IMO